docs/scarpet/api/BlocksAndWorldAccess.md

```diff
@@ -212,6 +212,15 @@ Causes a block to be harvested by a specified player entity. Honors player item
 tool if applicable. If the entity is not a valid player, no block gets destroyed. If a player is not allowed to break 
 that block, a block doesn't get destroyed either.
 
+### `weather(clear_ticks, rain_ticks, thunder)`,`weather(type)`
+
+Sets weather to clear for `ticks` ticks, after which it will rain for `second_ticks`
+ticks, and if `thunder` is true, it will thunder as well.
+
+If only 1 argument is specified, it must be either `'clear'`,`'rain'` or `'thunder'`, and will return true if the weather
+is in that state. NB: It can be thundering without there being a thunderstorm, there must be both thunder _and_ rain for
+there to be a thunderstorm.
```

Comment:
That is a little weird syntax. 
weather() -> returns 'clear', 'rain' or 'thunder' - current weather
weather(['clear', 'rain', 'thunder']) -> returns number of ticks remaining, or '0' if not in that mode
weather(mode, tick) -> same as /weather <mode> <ticks>

Not sure what would be the use of setting the ticks or clear weather with set ticks of rain weather. In these cases the app maker can set that directly calling weather again.

Reply:
ye but the function is built in like that.

Reply:
nvm ill do dat

Reply:
The function is built like that but it ain't used like that. API should not be restricted by contract of internal functions but adapt them to what makes sense - /weather command does exactly that.